README.md

```diff
@@ -85,6 +85,44 @@ ergonomic to write servers without dealing with sockets or serialization
 directly. Simply implement one of the generated traits, and you're off to the
 races! See the tarpc_examples package for more examples.
 
+## Example: Futures
+
+Here's the same server, implemented using `FutureService`.
+
+```rust
+#![feature(conservative_impl_trait, plugin)]
+#![plugin(tarpc_plugins)]
+
+extern crate futures;
+#[macro_use]
+extern crate tarpc;
+
+use tarpc::util::{FirstSocketAddr, Never};
+use tarpc::sync::Connect;
+
+service! {
+    rpc hello(name: String) -> String;
+}
+
+#[derive(Clone)]
+struct HelloServer;
+
+impl FutureService for HelloServer {
+    type HelloFut = futures::Finished<String, Never>;
```

Comment:
Agreed this could be documented better. WRT naming, my main concern is avoiding needless verbosity. Just for my understanding, what was the type you originally expected it to be?